### PR TITLE
[Account Numbers] Create searchable admin page of Column account numbers

### PR DESCRIPTION
**Pull request by rluodev**
## Summary of the problem

Ops had to ask engineers anytime they wanted to figure out which org owned an account number.

## Describe your changes

There is now a page for ops to find the org that owns an account number

Screenshot 2024-05-13 at 2 50 13 PM

Screenshot 2024-05-13 at 2 50 27 PM

Screenshot 2024-05-13 at 2 50 44 PM

Screenshot 2024-05-13 at 2 50 54 PM

Screenshot 2024-05-13 at 2 51 07 PM

Screenshot 2024-05-13 at 2 51 37 PM

### CODE REVIEW

**Comment by rluodev** [comment]
Note: due to `has_encrypted` I had to use blind index for this (which meant adding three migrations to add account_number_bidx, add index for said column, and backfill said column)

File: app/controllers/admin_controller.rb
Comment (cjdenio) [inline app/controllers/admin_controller.rb]:
```suggestion
    @account_numbers = relation.page(@page).per(@per).order("events.id desc")
```

this way it matches the sorting on `/admin/events`

File: app/views/admin/account_numbers.html.erb
Comment (cjdenio) [inline app/views/admin/account_numbers.html.erb]:
```suggestion
  <%= form.collection_select(:event_id, Event.reorder(Event::CUSTOM_SORT), :id, :admin_dropdown_description, { include_blank: "Select An Event", selected: @event_id }, { style: "max-width: 250px" }) %>
```

has no effect

File: app/views/admin/account_numbers.html.erb
Comment (cjdenio) [inline app/views/admin/account_numbers.html.erb]:
```suggestion
<%= form_with method: :get do |form| %>
```

File: app/views/admin/account_numbers.html.erb
Comment (cjdenio) [inline app/views/admin/account_numbers.html.erb]:
```suggestion
    <%= page_entries_info @account_numbers, entry_name: "Account Number" %>
```
`page_entries_info` automatically pluralizes the `entry_name`!

File: app/controllers/admin_controller.rb
Comment (garyhtou) [inline app/controllers/admin_controller.rb]: 👀

**Comment by rluodev** [comment]
aaaaaa

(Ok but actually I forgot I left that in there)

File: app/views/static_pages/_admin.html.erb
Comment (sampoder) [inline app/views/static_pages/_admin.html.erb]:
```suggestion
      <%= card_to "Account Numbers", account_numbers_admin_index_path, { badge: Column::AccountNumber.count, subtle_badge: true } %>
```
Any reason we can't just call it Account Numbers? I feel like that's cleaner

Comment (rluodev) [reply app/views/static_pages/_admin.html.erb]: fine by me!

File: app/controllers/admin_controller.rb
Comment (sampoder) [inline app/controllers/admin_controller.rb]: Do we want to fuzzy search?

**Review by sampoder — DISMISSED** [review]
Looks good to me! I just added a bit of styling to the top:

Screenshot 2024-05-14 at 12 12 14 PM

File: app/controllers/admin_controller.rb
Comment (rluodev) [reply app/controllers/admin_controller.rb]: would we want to use https://github.com/seamusabshere/fuzzy_match gem?

Comment (sampoder) [reply app/controllers/admin_controller.rb]: It's not exactly fuzzy searching but we can use the `ILIKE` operator like this:

```
scope :search, ->(q) { joins(:user).where("users.full_name ILIKE :query OR reimbursement_reports.name ILIKE :query", query: "%#{User.sanitize_sql_like(q)}%") }
```

That way if someone only puts in part of an account number it will still show up

**Comment by rluodev** [comment]
The issue with ilike is that with the has_encrypted attribute we’d have to
decrypt and load all account numbers into memory before searching them

File: app/controllers/admin_controller.rb
Comment (sampoder) [reply app/controllers/admin_controller.rb]: Yeah you're right, that's a bummer!

**Review by sampoder — APPROVED** [review]

**Review by hackliv — APPROVED** [review]

**Review by garyhtou — COMMENTED** [review]
🪄 good to merge after testing, and after another person has ran the migration on their computer!

**Review by sampoder — APPROVED** [review]
the migration worked on my computer

**Comment by rluodev** [comment]
i think it's waiting on someone from hcb-engineers group to approve

**Review by garyhtou — APPROVED** [review]

**Pull request merged**
